src/pybind/mgr/prometheus/module.py

```diff
@@ -692,23 +694,31 @@ def get_rbd_stats(self):
         # in the object name, the image pool is the pool where the object is
         # located.
 
-        # Parse rbd_stats_pools option, which is a comma or space separated
-        # list of pool[/namespace] entries. If no namespace is specifed the
-        # stats are collected for every namespace in the pool.
-        pools_string = self.get_localized_module_option('rbd_stats_pools', '')
         pools = {}
-        for p in [x for x in re.split('[\s,]+', pools_string) if x]:
-            s = p.split('/', 2)
-            pool_name = s[0]
-            if len(s) == 1:
-                # empty set means collect for all namespaces
-                pools[pool_name] = set()
-                continue
-            if pool_name not in pools:
-                pools[pool_name] = set()
-            elif not pools[pool_name]:
-                continue
-            pools[pool_name].add(s[1])
+        all_pools = strtobool(
+            self.get_localized_module_option('rbd_stats_all_pools', 'False'))
+        if all_pools:
+            osd_map = self.get('osd_map')
```

Comment:
@dillaman Don't you think `wait_for_latest_osdmap` could be useful here to ensure we have latest pools available?

Reply:
It should eventually get to consistent state, right? On the next periodic metrics scrape, it will see the new pool and refresh.